controllers/workspace/devworkspace_controller.go

```diff
@@ -72,6 +72,24 @@ type DevWorkspaceReconciler struct {
 	Scheme *runtime.Scheme
 }
 
+type errorReason interface {
+	error
+	Reason() string
+}
+
+func createErrorReason(err error, reason string) errorReason {
+	return errorReasonImpl{err, reason}
+}
+
+type errorReasonImpl struct {
+	error
+	reason string
+}
+
+func (e errorReasonImpl) Reason() string {
+	return e.reason
+}
+
```

Comment:
This `errorReason` interface was created to help set a failure reason when an error has occurred. For example:
https://github.com/devfile/devworkspace-operator/blob/ba8c5c1335339678a22a76ad8694546deabd61a3/controllers/workspace/status.go#L253
could fail due to an incorrect timeout value, or due to a timeout. 

If this `errorReason` is okay, I think functions like: https://github.com/devfile/devworkspace-operator/blob/ba8c5c1335339678a22a76ad8694546deabd61a3/controllers/workspace/devworkspace_controller.go#L286
could also benefit from it, since this function can fail for different reasons?